features/steps/helper.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	expectedLinksJob := strings.Replace(expectedResult["links: job"], "{bind_address}", f.Config.BindAddr, 1)
+	expectedLinksJob = strings.Replace(expectedLinksJob, "{id}", id, 1)
+
+	assert.Equal(&f.ErrorFeature, expectedLinksJob, links.Job)
+
+	expectedLinksSelf := strings.Replace(expectedResult["links: self"], "{bind_address}", f.Config.BindAddr, 1)
+	expectedLinksSelf = strings.Replace(expectedLinksSelf, "{id}", id, 1)
+	expectedLinksSelf = strings.Replace(expectedLinksSelf, "{task_name}", taskName, 1)
+
+	assert.Equal(&f.ErrorFeature, expectedLinksSelf, links.Self)
+	return nil
+}
+
+// checkValuesInJob is a utility function that can be called by a feature step in order to check that the values
+// of certain attributes, in a job, are all equal to the expected ones.
+func (f *SearchReindexAPIFeature) checkValuesInJob(expectedResult map[string]string, job models.Job) {
+	assert.Equal(&f.ErrorFeature, expectedResult["number_of_tasks"], strconv.Itoa(job.NumberOfTasks))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_completed"], job.ReindexCompleted.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_failed"], job.ReindexFailed.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_started"], job.ReindexStarted.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["state"], job.State)
+	assert.Equal(&f.ErrorFeature, expectedResult["total_search_documents"], strconv.Itoa(job.TotalSearchDocuments))
+	assert.Equal(&f.ErrorFeature, expectedResult["total_inserted_search_documents"], strconv.Itoa(job.TotalInsertedSearchDocuments))
+}
+
+// checkValuesInTask is a utility function that can be called by a feature step in order to check that the values
+// of certain attributes, in a task, are all equal to the expected ones.
+func (f *SearchReindexAPIFeature) checkValuesInTask(expectedResult map[string]string, task models.Task) {
+	assert.Equal(&f.ErrorFeature, expectedResult["number_of_documents"], strconv.Itoa(task.NumberOfDocuments))
+	assert.Equal(&f.ErrorFeature, expectedResult["task_name"], task.TaskName)
+}
+
+// readOutputMessages is a utility method to read the kafka messages that get sent to the producer's output channel
+func (f *SearchReindexAPIFeature) readOutputMessages() {
+	go func() {
+		for {
+			select {
+			case f.kafkaProducerOutputData <- <-f.KafkaMessageProducer.Channels().Output:
+				log.Println("read")
+			case <-f.quitReadingOutput:
+				return
+			}
+		}
+	}()
+}
+
+func readAndDeserializeKafkaProducerOutput(kafkaProducerOutputData <-chan []byte) (*models.ReindexRequested, error) {
+	reindexRequestedData := &models.ReindexRequested{}
+	reindexRequestedDataBytes := <-kafkaProducerOutputData
+
+	err := schema.ReindexRequestedEvent.Unmarshal(reindexRequestedDataBytes, reindexRequestedData)
+	if err != nil {
+		return &models.ReindexRequested{}, fmt.Errorf("failed to unmarshal reindex kafka message - err: %w", err)
```

Comment:
Surely we should return nil value for ReindexRequested?

```suggestion
		return nil, fmt.Errorf("failed to unmarshal reindex kafka message - err: %w", err)
```